src/Libraries/RevitNodes/Elements/Room.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using Autodesk.Revit.DB.Architecture;
+using DynamoServices;
+using Autodesk.DesignScript.Runtime;
+using Revit.GeometryConversion;
+using RevitServices.Persistence;
+using RevitServices.Transactions;
+using System.Collections.Generic;
+
+
+namespace Revit.Elements
+{
+    /// <summary>
+    /// Room Element
+    /// </summary>
+    [DynamoServices.RegisterForTrace]
+    public class Room : Element
+    {
+        #region Internal Properties
+
+        /// <summary>
+        /// Internal reference to the Revit Element
+        /// </summary>
+        internal Autodesk.Revit.DB.Architecture.Room InternalRevitElement
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Reference to the Element
+        /// </summary>
+        public override Autodesk.Revit.DB.Element InternalElement
+        {
+            get { return InternalRevitElement; }
+        }
+       
+        #endregion
+
+        #region Private constructors
+
+        /// <summary>
+        /// Create from existing element
+        /// </summary>
+        /// <param name="Room"></param>
+        internal Room(Autodesk.Revit.DB.Architecture.Room Room)
+        {
+            SafeInit(() => InitElement(Room));
+        }
+
+        /// <summary>
+        /// Create a new Room by Level and location
+        /// </summary>
+        /// <param name="level"></param>
+        /// <param name="location"></param>
+        /// <param name="name"></param>
+        /// <param name="number"></param>
+        private Room(Autodesk.Revit.DB.Level level, Autodesk.Revit.DB.XYZ location, string name, string number)
+        {
+            SafeInit(() => Init(level, location, name, number));
+        }
+
+        #endregion
+
+        #region Helpers for private constructors
+
+        /// <summary>
+        /// Set internal element
+        /// </summary>
+        private void InitElement(Autodesk.Revit.DB.Architecture.Room element)
+        {
+            InternalSetElement(element);
+        }
+
+        private void Init(Autodesk.Revit.DB.Level level, Autodesk.Revit.DB.XYZ location, string name, string number)
+        {
+            // Get document and open transaction
+            Autodesk.Revit.DB.Document document = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(document);
+
+            // Get existing room element if possible
+            var RoomElem = ElementBinder.GetElementFromTrace<Autodesk.Revit.DB.Architecture.Room>(document);
+
+            if (RoomElem == null)
+            {
+                // Create new Room element
+                RoomElem = document.Create.NewRoom(level, new Autodesk.Revit.DB.UV(location.X, location.Y));
+            }
+            else
+            {
+                // Update Location only
+                Autodesk.Revit.DB.LocationPoint point = (Autodesk.Revit.DB.LocationPoint)RoomElem.Location;
```

Comment:
@sharadkjaiswal , good point. haven't found a way to do this via API at the moment. the properties Level and LevelID are readonly. I could create a new room and delete the old one but this might affect existing properties.